File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    num = len(p)
    p_num = int(p) 
    
    for i in range(len(t) - num + 1):
        temp = t[i:i + num]
        if int(temp) <= p_num:  
            answer += 1
    
    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int count=0;
        int len=p.length();
        long pNum=Long.parseLong(p);
        
        for(int i=0;i<t.length()-len+1;i++){
            Long num=Long.parseLong(t.substring(i,i+len));
            if(num<=pNum){
                count++;
            }
        }
        return count;
    }
}